tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -700,6 +732,7 @@ def unlock(self):
         self.marionette.import_script(self.lockscreen_atom)
         self.marionette.switch_to_frame()
         result = self.marionette.execute_async_script('GaiaLockScreen.unlock()')
+        GaiaData(self.marionette).set_setting('lockscreen.enabled', False)
```

Comment:
Similar to above, this leaves a side effect behind. It also assumes that lock has been called before unlock, I think.

Both of these should probably be rethought with the idea of an uncertain setting state in mind. They really should cache the old value and restore it, for one thing, rather than leaving side effects.